Engineering/Frontend/Window and iframe communcation.md

```diff
@@ -0,0 +1,91 @@
+---
+tags: frontend, engineering/frontend, iframe, postMessage, technique
+author: Pham Duc Thanh
+github_id: zlatanpham
+date: 2023-08-20
+---
+
+When working with web development, communicating between a main window and an embedded iframe is often necessary for various tasks, such as content creation or widget implementation. This communication can occur between windows and iframes with the same origin or across different origins.
+
+## Same Origin Communication
+
+"Same origin" refers to the scenario where two URLs have the same protocol, domain, and port. For instance, the following URLs have the same origin:
```

Comment:
Maybe it's not different at all and `protocol` is still fine but technically in a URLs, it should be `scheme` instead of `protocol`

Reply:
I believe the difference lies in the usage of language. Some sources use the term ‘protocol’, while others prefer ‘scheme’. Although I believe that ‘protocol’ may be more commonly understood, if correctness is in question, I am willing to update my terminology at a later time.

![image](https://github.com/dwarvesf/brain/assets/12707960/f7bf7369-8265-45ab-b752-386127703e96)